utils/build_and_install.sh

```diff
@@ -728,6 +757,33 @@ check_and_clone_git_dependency()
     echo_log "-- [OK] Found local repository for ${git_dep} in ${repo_dir}"
 
     if check_clean_work_tree; then
+      # Ensure that the remote is correct
+      echo "-- Checking remote URL"
+      remote="`git remote get-url origin`"
+      if [ "$remote" != "$git_dep_uri" ]
+      then
+        echo_log "-- The remote has changed for $repo, previous remote: $remote, desired remote $git_dep_uri."
+
+        if $ASK_USER_INPUT
+        then
+          read -r -p "--> Would you like to update to the new remote (warning this will overwrite the local master branch)? [y/N] " response
+          if [[ "$response" =~ ^([yY][eE][sS]|[yY])$ ]]
+          then
+            echo_log "-- Updating the remote url to $git_dep_uri"
+            git remote set-url origin $git_dep_uri
+            git remote update origin
+            exit_if_error "Failed to update the remote uri to $git_dep_uri"
+            git fetch origin
+            exit_if_error "Failed to fetch the remote uri to $git_dep_uri"
+            git reset --hard origin/master
+            exit_if_error "Failed to update to the the remote's master branch (new remote: $git_dep_uri)"
+          else
+            echo_log "-- Installation stopped because the locate remote $remote does not match the local remote uri ."
+            echo_log "   Please manually update the remote to $git_dep_uri and rerun the script."
+            exit_failure
+          fi
+        fi
+      fi
```

Comment:
This is annoying for those who use the script but at some point switch their repo to point at the pushable version

I suggest we simply don't check that

Reply:
Agreed this can be annoying, in it's current form it's a bit too intrusive as it'll match exactly the remote URI. But it can also lead to failure not to do it in the case where the remote has changed. This was notably the case for `mc_openrtm` that used to be on gite, but that we moved to github only, and deprecated the repository on gite.

However, we could simply do this check on a case-by-case basis instead (we know which remotes got moved, so we could explicitly check for those).

Reply:
I've relaxed this check, it'll now only match if one remote contains `github` and the other contains `gite`. That'll catch the `mc_openrtm` case and offer to update the remote and corresponding local branch to github, but leave all the other remotes unchanged.